### PR TITLE
[openwebnet] add sendMessage rule actions to send generic OWN messages on the BUS

### LINKED ISSUE #10431: [openwebnet] allow sending generic OWN messages from rules
Allow sending generic OWN frames from OH rules. This should be useful to activate several actions from rules even if a specific WHO is not supported in the binding.

### PR DESCRIPTION
**Pull request by mvalla**
This PR adds two new rule actions to allow sending generic OpenWebNet messages from Rules.
The new actions are:

- `Boolean sendMessage(String message)` returns a `Boolean` = `true` if the `message` (OpenWebNet frame) was successfully sent via the gateway, `false` otherwise.
- `Map<String, Object> sendMessageWithResponse(String message)` same as previous one, but returns a `Map<String, Object>` with following keys:
    - `success`: a `Boolean` = `true` if the `message` was sent successfully
    - `responseMessages`: a `List<String>` object containing all returned frames as response to command sent

`openwebnet4j` library has been updated to 0.14.0 to allow using `DefaultMessage` to send messages not yet supported by the binding.

Fixes #10431.

### CODE REVIEW

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/actions/OpenWebNetBridgeActions.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        if (gw != null && gw.isConnected()) {
+            try {
+                Response res = gw.send(msg);
+                logger.debug("sent message {} to gateway. Response: {}.", msg, res.getResponseMessages());
+                responseMap.put("success", res.isSuccess());
+                List<String> resultList = res.getResponseMessages().stream().map(rm -> rm.getFrameValue())
+                        .collect(Collectors.toList());
+                responseMap.put("responseMessages", resultList);
+                return responseMap;
+
```
Comment (lsiepel) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/actions/OpenWebNetBridgeActions.java:114]:
```suggestion
```

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/actions/OpenWebNetBridgeActions.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            logger.warn("openwebnet skipping sendMessage for bridge {}: gateway is not connected.",
+                    handler.getThing().getUID());
+            return responseMap;
+        }
+    }
+
+    // legacy delegate methods
+    public static Boolean sendMessage(@Nullable ThingActions actions, String message) {
+        if (actions instanceof OpenWebNetBridgeActions) {
+            return ((OpenWebNetBridgeActions) actions).sendMessage(message);
```
Comment (lsiepel) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/actions/OpenWebNetBridgeActions.java:129]:
```suggestion
        if (actions instanceof OpenWebNetBridgeActions openwebnetBridgeActions) {
            return openwebnetBridgeActions.sendMessage(message);
```

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/actions/OpenWebNetBridgeActions.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        } else {
+            throw new IllegalArgumentException("Instance is not an OpenWebNetBridgeActions class.");
+        }
+    }
+
+    public static Map<String, Object> sendMessageWithResponse(@Nullable ThingActions actions, String message) {
+        if (actions instanceof OpenWebNetBridgeActions) {
+            return ((OpenWebNetBridgeActions) actions).sendMessageWithResponse(message);
```
Comment (lsiepel) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/actions/OpenWebNetBridgeActions.java:137]:
```suggestion
        if (actions instanceof OpenWebNetBridgeActions) {
            return ((OpenWebNetBridgeActions) actions).sendMessageWithResponse(message);
```

**Review by lsiepel — CHANGES_REQUESTED** [review]
Thanks, left some minor comments.

**Comment by mvalla** [comment]
> Thanks, left some minor comments. Can you also fix the warnings from here https://github.com/openhab/openhab-addons/actions/runs/8860857344/artifacts/1453450970

Done.

> Besides the code review, why is there a need for sending generic message ? Is there a reason some datapoints/communication cannot be modelled as thing/channel within the openHAB domain ?

Until all automation areas will be supported by the binding (few complex ones like Video Door entry or Room Audio control are not yet handled), it is useful to let end-user send generic (=not supported) messages via the binding.
This empowers openHAB to work as generic message sender to the BTicino MyHOME BUS.

**Comment by Nils-github** [comment]
@mvalla : Thanks for this nice feature!
Could it be that you forgot to release the dependency: openwebnet4j version 0.14.0 ?

**Comment by lsiepel** [comment]
> @mvalla : Thanks for this nice feature! Could it be that you forgot to release the dependency: openwebnet4j version 0.14.0 ?

No, https://mvnrepository.com/artifact/io.github.openwebnet4j/openwebnet4j
If you refer to the github repo, please create an issue at that repository.

**Pull request merged**